ivy/functional/backends/jax/general.py

```diff
@@ -208,7 +208,7 @@ def inplace_update(
     keep_input_dtype: bool = False,
 ) -> ivy.Array:
     if ivy.is_array(x) and ivy.is_array(val):
-        if ensure_in_backend:
+        if ensure_in_backend or (ivy.is_native_array(x) and not ivy.get_array_mode()):
```

Comment:
One thing I realized, we don't have to check for `ivy.get_array_mode()` since `x` can only be native if this mode was disabled 😅.
So we can remove this check  here and in other backends as well.

Reply:
Yes, you are right, I pushed the new changes. I hope, Now this will work fine 🌝